src/dx/presence/builders/here_now.rs

```diff
@@ -0,0 +1,187 @@
+    #[builder(
+        field(vis = "pub(in crate::dx::presence)"),
+        setter(strip_option, into),
+        default = "vec![]"
+    )]
+    pub(in crate::dx::presence) channels: Vec<String>,
+
+    /// Channel groups for which to retrieve occupancy information.
+    #[builder(
+        field(vis = "pub(in crate::dx::presence)"),
+        setter(into, strip_option),
+        default = "vec![]"
+    )]
+    pub(in crate::dx::presence) channel_groups: Vec<String>,
+
+    /// Whether to include UUIDs of users subscribed to the channel(s).
+    #[builder(
+        field(vis = "pub(in crate::dx::presence)"),
+        setter(strip_option),
+        default = "true"
+    )]
+    pub(in crate::dx::presence) include_user_id: bool,
+
+    /// Whether to include state information of users subscribed to the channel(s).
+    #[builder(
+        field(vis = "pub(in crate::dx::presence)"),
+        setter(strip_option),
+        default = "false"
+    )]
+    pub(in crate::dx::presence) include_state: bool,
+}
+
+impl<T, D> HereNowRequestBuilder<T, D> {
+    /// Validate user-provided data for request builder.
+    ///
+    /// Validator ensure that list of provided data is enough to build valid
+    /// set state request instance.
+    fn validate(&self) -> Result<(), String> {
+        let groups_len = self.channel_groups.as_ref().map_or_else(|| 0, |v| v.len());
+        let channels_len = self.channels.as_ref().map_or_else(|| 0, |v| v.len());
+
+        builders::validate_configuration(&self.pubnub_client).and_then(|_| {
+            if channels_len == groups_len && channels_len == 0 {
+                Err("Either channels or channel groups should be provided".into())
+            } else {
+                Ok(())
+            }
+        })
+    }
+
+    /// Build [`SetStateRequest`] from builder.
+    fn request(self) -> Result<HereNowRequest<T, D>, PubNubError> {
+        self.build()
+            .map_err(|err| PubNubError::general_api_error(err.to_string(), None, None))
+    }
+}
+
+impl<T, D> HereNowRequest<T, D> {
+    /// Create transport request from the request builder.
+    pub(in crate::dx::presence) fn transport_request(
+        &self,
+    ) -> Result<TransportRequest, PubNubError> {
+        let sub_key = &self.pubnub_client.config.subscribe_key;
+        let mut query: HashMap<String, String> = HashMap::new();
+
+        // Serialize list of channel groups and add into query parameters list.
+        url_encoded_channel_groups(&self.channel_groups)
+            .and_then(|channel_groups| query.insert("channel-group".into(), channel_groups));
+
+        self.include_state
+            .then(|| query.insert("state".into(), "1".into()));
+
+        (!self.include_user_id).then(|| {
+            query.insert("disable_uuids".into(), "1".into());
+        });
+
+        query.insert(
+            "uuid".into(),
+            self.pubnub_client.config.user_id.deref().clone(),
+        );
```

Comment:
This one did not require because it is included by middleware with required encoding.

Reply:
fixed